### DISCUSSION TITLE
Is there any plans to support Snowflake as another backend?

### DISCUSSION THREAD

**Discussion by zhyzhkea** [moj-analytical-services/splink#1314] [Q&A] [answered]
I would like to run splink using Snowpark Virtual Warehouse using Snowflake compute nodes but it seems that duckdb has not supported by Snowpark. 

Thanks

**Comment by RobinL** [comment, marked as answer]
Hello.   I'm afraid there are no current plans to support Snowflake, partly because we do not have access to Snowflake for testing, and also we don't have capacity on the team to write the requisite adaptors.

We would consider accepting a PR, but if you (or someone else) wanted to work on this, please ask first because we would need to consider whether we have capacity to maintain the code

**Reply by upendrasingh1** [reply]
I would like to work on it and contribute for Snowflake.

**Reply by RobinL** [reply]
Hello.  Thank you for the offer  We're super busy at the moment so probably don't have time to support you with this at the moment.

However, we are working on Splink 4 [here](https://github.com/moj-analytical-services/splink/tree/splink4_dev).  One importatn feature is the new database api, which formalises the contract between Splink and the database backend.

It means that it's possible to create a new database api class for a new backend.  So you could do that without even needing to modify the Splink codebase.

If you feel confident in implementing a Snowflake backend for Splink 4, we'd certainly consider accepting a PR, but would probably be unable to matain it (given that we don't have access to snowflake)

Work on the new database api is here:
https://github.com/moj-analytical-services/splink/blob/splink4_dev/splink/database_api.py

And you can see examples of how it is used in the tutorials e.g here https://github.com/moj-analytical-services/splink/blob/splink4_dev/docs/demos/tutorials/05_Predicting_results.ipynb

**Reply by upendrasingh1** [reply]
I am pretty confident that support for snowflake can be added. I have working code for snowflake for latest version on splink on my system but have not used Splink 4 apis for development. I added support for snowflake in the same fashion as we would have in 3.X.X versions of splink. I can have a look at Splink 4 database api and work on that. Given the case that it is possible, where would u recommend I add my changes, in Splink 4 or Splink 3?

**Reply by upendrasingh1** [reply]
I also understand the maintaining part....I will try to maintain it for as long as I can.
